### PR TITLE
Calculate distance

### PR DESCRIPTION
**Pull request by ducbet**
Link: none
Effect: MainActivity: Calculate distance again

### CODE REVIEW

File: app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java [hunk 1]
```diff
@@ -1,9 +1,14 @@
 package com.framgia.feastival.screen.main;
 
+import android.util.Log;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: ERROR: Unused import: android.util.Log. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java [hunk 2]
```diff
@@ -82,4 +88,15 @@ public void onComplete() {
...
     }
+
+    @Override
+    public void calculateDistance(LatLng location, double radius, RestaurantsResponse response) {
+        List<Restaurant> list = response.getList();
+        for (int i = list.size() - 1; i >= 0; i--) {
+            if (((MainViewModel) mViewModel).getDistance(location,
+                new LatLng(list.get(i).getLatitude(), list.get(i).getLongtitude())) > radius) {
+                list.remove(i);
+            } else break;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: WARNING: 'else' construct must use '{}'s.

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java [hunk 3]
```diff
@@ -10,6 +10,7 @@
 import android.support.annotation.NonNull;
 import android.support.design.widget.BottomSheetBehavior;
 import android.support.v4.app.ActivityCompat;
+import android.util.Log;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: ERROR: Unused import: android.util.Log. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java [hunk 4]
```diff
@@ -75,7 +76,7 @@ public MainViewModel(Context context) {
 
     private void zoomInMyPositonAutomaticly() {
         if (mMap != null && isNeedInMyLocation) {
-            mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(mMyLocation, 14));// 0-18
+            mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(mMyLocation, 14));// 0-18
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: ERROR: ';' is not followed by whitespace.

File: app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java (hunk 1 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java (hunk 2 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: WARNING: [IfElseStmtsMustUseBraces] Avoid using if...else statements without curly braces
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/braces.html#IfElseStmtsMustUseBraces

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java (hunk 3 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/res/drawable/ic_house.xml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<vector android:height="24dp" android:viewportHeight="486.196"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_house.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_house` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/ic_keyboard_arrow_down_white_24px.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_keyboard_arrow_down_white_24px.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_keyboard_arrow_down_white_24px` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/ic_keyboard_arrow_up_white_24px.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_keyboard_arrow_up_white_24px.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_keyboard_arrow_up_white_24px` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/ic_phone_call.xml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<vector android:height="24dp" android:viewportHeight="578.106"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_phone_call.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_phone_call` appears to be unused
 Unused resources make applications larger and slow down builds.

Comment (ducbet) [inline app/src/main/res/drawable/ic_phone_call.xml]: em up trước 1 số ảnh về sau sẽ dùng ạ

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java [hunk 9]
```diff
@@ -197,7 +197,7 @@ private void markNearbyRestaurants(RestaurantsResponse restaurantsResponse) {
         }
     }
 
-    private double getDistance(LatLng latLngA, LatLng latLngB) {
+    public double getDistance(LatLng latLngA, LatLng latLngB) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: Hàm này viết bên presenter

File: app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java [hunk 10]
```diff
@@ -61,13 +64,14 @@ public void onComplete() {
...
     @Override
-    public void getRestaurants(LatLng location, double radius) {
+    public void getRestaurants(final LatLng location, final double radius) {
         Disposable disposable = mRestaurantRepository.getRestaurants(location, radius)
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribeWith(new DisposableObserver<RestaurantsResponse>() {
                 @Override
                 public void onNext(@NonNull RestaurantsResponse response) {
+                    calculateDistance(location, radius, response);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: Sao lại tính ntn?

Comment (ducbet) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: vì bên server tính lệch khoảng 150% nên về app phải lọc lại 1 lần để hiển thị ạ

Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainPresenter.java]: Tốn time quá. Báo server tính lại

**Pull request merged**